src/popup/components/TransactionList.vue

```diff
@@ -1,7 +1,16 @@
 <template>
   <div class="transaction-list">
+    <div
+      v-if="!showFilters && displayMode.filter !== 'all'"
+      class="search-bar-wrapper"
+    >
+      <InputSearch
+        v-model="search"
+        :placeholder="$t('pages.recentTransactions.searchPlaceholder')"
+      />
+    </div>
     <Filters
-      v-if="showFilters"
+      v-if="showFilters || displayMode.filter !== 'all'"
```

Comment:
I'd suggest moving `displayMode.filter !== 'all'` to computed not to repeat yourself